csrc/transformer/inference/csrc/pt_binding.cpp

```diff
@@ -305,7 +312,15 @@ void attention_unfused(T* prev_key_cont,
     float layer_scale = alibi.sizes().size() > 1 ? std::max(1, layer_id) : 1.0;
     float alpha = norm_factor * norm_factor / layer_scale;
     float gemm_beta = 0.0;
-    T* workspace = (T*)output + bsz * seq_len * heads * k;
+    T* workspace;
+    if (seq_len == 1) {
+        workspace = (T*)output + bsz * seq_len * heads * k;
+    } else {
+        // If we are doing the prompt, switch to the tail workspace
+        T* scratch = (T*)Context::Instance().GetWorkSpace();
+        workspace = scratch + (Context::Instance().get_workspace_size() / sizeof(T)) -
+                    heads * seq_len * seq_len;
```

Comment:
@cmikeh2, do we want to add bsz here too, since we might get a batched input?

Reply:
Yeah, this is primed to cause a segfault on any batched input.